shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/resources/CamelIntegrationStatus.java

```diff
@@ -0,0 +1,44 @@
+package com.redhat.service.smartevents.shard.operator.v2.resources;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.redhat.service.smartevents.shard.operator.core.resources.Condition;
+import com.redhat.service.smartevents.shard.operator.core.resources.ConditionStatus;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CamelIntegrationStatus {
+
+    private Set<Condition> conditions = new HashSet<>();
+
+    public Set<Condition> getConditions() {
+        return conditions;
+    }
+
+    public void setConditions(Set<Condition> conditions) {
+        this.conditions = conditions;
+    }
+
+    @JsonIgnore
+    public final boolean isReady() {
+        return conditionsAreTrue("IntegrationKitAvailable", "IntegrationPlatformAvailable", "KnativeServiceAvailable", "KnativeServiceReady");
```

Comment:
@lucamolteni Constants for the Strings here?

Reply:
We've been trying to keep `Condition` constants [here](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/DefaultConditions.java) so they're reusable from both the `manager` and `operator`.